sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -68,6 +124,126 @@ public String getPassword() {
         return this.password;
     }
 
+    /**
+     * @return the hosts that bypass the proxy.
+     */
+    public String getNonProxyHosts() {
+        return this.nonProxyHosts;
+    }
+
+    /**
+     * Attempts to load a proxy from the environment.
+     *
+     * <p>
+     * Environment configurations are loaded in this order:
+     * <ol>
+     *     <li>Azure HTTPS</li>
+     *     <li>Azure HTTP</li>
+     *     <li>Java HTTPS</li>
+     *     <li>Java HTTP</li>
+     * </ol>
+     *
+     * Azure proxy configurations will be preferred over Java proxy configurations as they are more closely scoped to
+     * the purpose of the SDK. Additionally, more secure protocols, HTTPS vs HTTP, will be preferred.
+     *
+     * <p>
+     * {@code null} will be returned if no proxy was found in the environment.
+     *
+     * @param configuration The {@link Configuration} that is used to load proxy configurations from the environment.
+     * If {@code null} is passed then {@link Configuration#getGlobalConfiguration()} will be used. If
+     * {@link Configuration#NONE} is passed {@link IllegalArgumentException} will be thrown.
+     * @return A {@link ProxyOptions} reflecting a proxy loaded from the environment, if no proxy is found {@code null}
+     * will be returned.
+     * @throws IllegalArgumentException If {@code configuration} is {@link Configuration#NONE}.
+     */
+    public static ProxyOptions loadFromConfiguration(Configuration configuration) {
```

Comment:
I wonder if we should call this `fromConfiguration`, that feels a little more Java idiomatic

Reply:
Done